### PR TITLE
Add commentary for prepared statement and related code

### PR DESCRIPTION
**Pull request by OpatrilPeter**
Documentation for PreparedStatement machinery.
Things still requiring review is annotated by !!!!!.
Correct generation of documentation by Doxygen is not properly tested yet.

### CODE REVIEW

File: include/superior_mysqlpp/prepared_statement.hpp [hunk 1]
```diff
@@ -28,6 +28,17 @@
 
 namespace SuperiorMySqlpp
 {
+    /**
+     * @brief Bindings of storage for either parameters (inputs) or results (outputs) of SQL queries.
+     * Bindings class encapsulates the storage of heterogenous data needed for SQL parameters and results.
+     * The binding info itself is stored in std::tuple @a data, auxiliary binding information is stored in array of
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statement.hpp]: Actually bindings storage is stored in std::tuple data, isn't it?

Comment (majcl) [reply include/superior_mysqlpp/prepared_statement.hpp]: Isn't it possible to ref data(in "...std::tuple @a data, auxili..." by #data or something similar, instead of manual highliting with \@a?

File: include/superior_mysqlpp/prepared_statement.hpp [hunk 2]
```diff
@@ -28,6 +28,17 @@
...
+     * MYSQL_BIND structures used from underlying C mysql client -- member @a bindings.
+     * @tparam IsParamBingngs - Special flag for template specialization.
+     *                          If Bindings is used for query parameter, it is true.
+     *                          If Bindings is used for query result, it is false.
+     * @tparam Types - Parameter pack of types forming storage of individual fields.
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statement.hpp]: storage *for* individual... ?

File: include/superior_mysqlpp/prepared_statement.hpp [hunk 3]
```diff
@@ -40,8 +51,12 @@ namespace SuperiorMySqlpp
 
 
     public:
-        /* Note: using just 'bindings{}' triggers (premature) warning in GCC 4.9.2,
-         * see discussion at https://stackoverflow.com/questions/31271975/why-can-i-initialize-a-regular-array-from-but-not-a-stdarray */
+        /*
+         * Bindings constructor.
+         * @params args Args are used for tuple initialization.
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statement.hpp]: Probably should be specified, that args are used as a storage initialization.

File: include/superior_mysqlpp/prepared_statement.hpp [hunk 4]
```diff
@@ -56,19 +71,47 @@ namespace SuperiorMySqlpp
...
+    /*
+     * @brief Static MySQL prepared statement.
+     * Static prepared statements needs to know the parameter bindings.
+     * Two advantages compared to using #Query - binary communication with server
+     * and stronger type safety guarantees.
+     * Created by constructor or Connection::makePreparedStatement.
+     * @tparam ResultBindings Specialization of #ResultBindings.
+     * @tparam ParamBindings Specialization of #ParamBindings.
+     * @tparam storeResult If true, fetches results in "store mode" -- whole result set in one operation.
+     *                     If false, fetches results in "use mode", incrementally.
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statement.hpp]: This two lines should probably have similar formating ;-)

File: include/superior_mysqlpp/prepared_statement.hpp [hunk 5]
```diff
@@ -56,19 +71,47 @@ namespace SuperiorMySqlpp
...
+     *                     Both modes offers some additional unique methods.
+     * @tparam validateMode Defines validation strategy, i.e. how strict type control is.
+     *                      If stronger class of conversion is required, it leads to error.
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statement.hpp]: conversion class

File: include/superior_mysqlpp/prepared_statement.hpp [hunk 6]
```diff
@@ -56,19 +71,47 @@ namespace SuperiorMySqlpp
...
+     * @tparam warnMode Defines level of conversion that triggers warning, but is accepted.
+     * @tparam ignoreNullable If true, one can store #Nullable<T> in storage of type T. If the nullable is empty,
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statement.hpp]: I would rephrase - if field is actually NULL, the storage will remain unchanged...

File: include/superior_mysqlpp/prepared_statement.hpp [hunk 7]
```diff
@@ -139,6 +207,16 @@ namespace SuperiorMySqlpp
             InitializeResult(resultBindings.data);
         }
 
+        /**
+         * @brief Constructor for PreparedStatement.
+         * This overload is for useful when Connection is available.
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statement.hpp]: Link to Connection

File: include/superior_mysqlpp/prepared_statement.hpp [hunk 8]
```diff
@@ -56,19 +71,47 @@ namespace SuperiorMySqlpp
...
+     * @tparam ResultBindings Specialization of #ResultBindings.
+     * @tparam ParamBindings Specialization of #ParamBindings.
+     * @tparam storeResult If true, fetches results in "store mode" -- whole result set in one operation.
+     *                     If false, fetches results in "use mode", incrementally.
+     *                     Both modes offers some additional unique methods.
+     * @tparam validateMode Defines validation strategy, i.e. how strict type control is.
+     *                      If stronger class of conversion is required, it leads to error.
+     * @tparam warnMode Defines level of conversion that triggers warning, but is accepted.
+     * @tparam ignoreNullable If true, one can store SuperiorMySqlpp::Nullable<T> in storage of type T. If the nullable is empty,
+     *                        nothing will be assigned.
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statement.hpp]: What does mean if nullable is empty?

File: include/superior_mysqlpp/prepared_statement.hpp [hunk 9]
```diff
@@ -185,42 +274,72 @@ namespace SuperiorMySqlpp
...
+         * Accessor of tuple containing storage of result bindings.
+         * Const variant.
+         */
         const auto& getResult() const noexcept
         {
             return resultBindings.data;
         }
 
+        /**
+         * Convenience function to replace current query params.
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statement.hpp]: What does it exactly mean replace current query params? And what about connection to updateParamsindings? Please explain this more thoroughly.

File: include/superior_mysqlpp/prepared_statement.hpp [hunk 10]
```diff
@@ -185,42 +274,72 @@ namespace SuperiorMySqlpp
...
+         * @param args Forwarded new data,
+         */
         template<typename... Args>
         void setParams(Args&&... args)
         {
             paramsBindings.data = std::forward_as_tuple(std::forward<Args>(args)...);
         }
 
+        /**
+         * Returns size of parameterBindings.
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statement.hpp]: Is it really size or count of binded parameters?

File: include/superior_mysqlpp/prepared_statement.hpp [hunk 11]
```diff
@@ -185,42 +274,72 @@ namespace SuperiorMySqlpp
...
+         */
         static constexpr auto getParamsCount() noexcept
         {
             return decltype(paramsBindings)::kArgumentsCount;
         }
 
+        /**
+         * Returns size of resultBindings.
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statement.hpp]: Is it really size or count of binded results?

File: include/superior_mysqlpp/prepared_statement.hpp [hunk 12]
```diff
@@ -185,42 +274,72 @@ namespace SuperiorMySqlpp
...
+         */
         static constexpr auto getResultCount() noexcept
         {
             return decltype(resultBindings)::kArgumentsCount;
         }
 
+        /**
+         * @brief Executes prepared statement.
+         * This method actually performs given query, after validating its metadata.
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statement.hpp]: Isn't it the other way around? First performs query and then validates actual result metadata with result bindings?

File: include/superior_mysqlpp/prepared_statements/binding_types.hpp [hunk 13]
```diff
@@ -25,10 +25,19 @@ namespace SuperiorMySqlpp
             Timestamp,
         };
 
-
+        /**
+         * This boolean trait describes if type T can be uaed as a storage for
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statements/binding_types.hpp]: typo: uaed => used

File: include/superior_mysqlpp/prepared_statements/binding_types.hpp [hunk 14]
```diff
@@ -25,10 +25,19 @@ namespace SuperiorMySqlpp
...
+         * a given binding type (of enum type #BindingTypes).
+         * This is a variant for query parameters and defaults to false.
+         */
         template<BindingTypes, typename T>
         struct CanBindAsParam : std::false_type {};
 
+        /**
+         * This boolean trait describes if type T can be uaed as a storage for
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statements/binding_types.hpp]: typo uaed => used

File: include/superior_mysqlpp/prepared_statements/default_initialize_result.hpp [hunk 15]
```diff
@@ -10,8 +10,10 @@
 
 namespace SuperiorMySqlpp
 {
-    /*
-     * initializeResultItem
+    /**
+     * Generic template and default implementation of initialization for one field of result binding.
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statements/default_initialize_result.hpp]: This needs to be explained more throroughly. Where and how and why this is used.

File: include/superior_mysqlpp/prepared_statements/get_binding_type.hpp [hunk 16]
```diff
@@ -16,27 +16,44 @@ namespace SuperiorMySqlpp
 {
     namespace detail
     {
+        /**
+         * Mapping of native C++ types to fitting field type
+         * @return FieldType enumeration of matching type
+         * !!!!! This code is incorrect - sql types are exact size, C++ are architecture dependent
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statements/get_binding_type.hpp]: Correct note, but, is underlying C mysql client even available for different architectures?

File: include/superior_mysqlpp/prepared_statements/get_binding_type.hpp [hunk 17]
```diff
@@ -16,27 +16,44 @@ namespace SuperiorMySqlpp
 {
     namespace detail
     {
+        /**
+         * Mapping of native C++ types to fitting field type
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statements/get_binding_type.hpp]: I would add to *mysql* field type. Also somewhere should be added, that it is preffered to use *mysql* types instead of native types, wherever possible.

File: include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp [hunk 18]
```diff
@@ -14,14 +14,23 @@
...
+         * Trait specialization saying that std::string can be used as parameter
+         * for string field type.
+         */
         template<>
         struct CanBindAsParam<BindingTypes::String, std::string> : std::true_type {};
 
 
-        /*
-         * Common bindings
+        /**
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp:42]: It would be helpful here, to have more thorough description, what is done in the function and that this is only internal part of actual initialization(because Nullable<T> must have also initialize its "content" T)

File: include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp [hunk 19]
```diff
@@ -45,6 +54,9 @@ namespace SuperiorMySqlpp
             binding.is_unsigned = std::is_unsigned<PureType_t>()? true : false;
         }
 
+        /**
+         * Param binding initialization specialization for floats.
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp:72]: Please mention in doc also long double exception and possibly the reason.

File: include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp [hunk 20]
```diff
@@ -14,14 +14,23 @@
 
 namespace SuperiorMySqlpp
 {
+    /*
+     * Objective of this whole file is proper initialization of bindings.
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp:18]: Also please describe a little how it is actually designed. It goes a long way, when someone needs to add something. Also when and how are these initializations called.

File: include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp [hunk 21]
```diff
@@ -173,48 +229,68 @@ namespace SuperiorMySqlpp
...
-         * Initialize binding implementation
+        /**
+         * Initializes individual binding.
+         * Generic template, actual implementation is in specializations.
          */
         template<bool isParamBinding>
         struct InitializeBindingImpl;
 
+        /**
+         * Specialization of #InitializeBindingImpl for query parameters.
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp:309]: Explain please the specialization by true/false template argument

File: include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp [hunk 22]
```diff
@@ -270,8 +355,10 @@ namespace SuperiorMySqlpp
         };
 
 
-        /*
-         * Initialize Bindings
+        /**
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp:375]: This is probably the only "importantly-visible exported" method from this module. Please describe also in java doc, how does it works.

File: include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp [hunk 23]
```diff
@@ -68,23 +94,39 @@ namespace SuperiorMySqlpp
             }
 
         protected:
+            /**
+             * This method implements the store() specialized behavior.
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp]: What does actually mean "specialized behaviour"?

File: include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp [hunk 24]
```diff
@@ -68,23 +94,39 @@ namespace SuperiorMySqlpp
...
+             */
             void storeOrUse()
             {
                 this->statement.storeResult();
             }
         };
 
+        /**
+         * Specialization of StatementBase where statement results are "used" - not
+         * fetched all at one.
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp]: The code design, how prepared statement distinguishes between *store* and *use* results should be explained somewhere.

File: include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp [hunk 25]
```diff
@@ -68,23 +94,39 @@ namespace SuperiorMySqlpp
...
+             * As this is the default mode for mysql client, nothing needs to be done.
+             */
             void storeOrUse() const
             {
             }
 
+            /**
+             * Returns the amount of fetched rows.
+             * @return Amount of already fetched rows.
+             * !!!!! Underlying mysql_stmt_num_rows does NOT gurantee such behavior!
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp]: This returns already fetched rows, not total result rows, or even this lighter guarantee isn't right?

File: include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp [hunk 26]
```diff
@@ -155,7 +203,11 @@ namespace SuperiorMySqlpp
             }
 
         public:
-
+            /**
+             * Returns metadata for the statement result.
+             * There metadata are retrieveable even before execute().
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp:219]: Are you sure they are available before execute()?

File: include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp [hunk 27]
```diff
@@ -192,45 +254,71 @@ namespace SuperiorMySqlpp
                 return ok;
             }
 
-            /**
-             * TODO: Statement.fetchColumn is void and can throw
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp:196]: Why this comment is deleted?

File: include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp [hunk 28]
```diff
@@ -192,45 +254,71 @@ namespace SuperiorMySqlpp
...
+             * @return void
+             * !!!!! Should return void, this form is misleading.
+             */
             auto sendLongData(unsigned int paramNumber, const std::string& data)
             {
                 return this->statement.sendLongData(paramNumber, data);
             }
 
+            /**
+             * Sets how many rows shall be prefetched in single fetch call.
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp:305]: Does it have meaning for both *use* and *store* results?

File: include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp [hunk 29]
```diff
@@ -192,45 +254,71 @@ namespace SuperiorMySqlpp
...
+             */
             unsigned long getPrefetchRowCount()
             {
                 unsigned long result;
                 this->statement.getAttribute(LowLevel::DBDriver::Statement::Attributes::prefetchRows, result);
                 return result;
             }
 
+            /**
+             * If related attribute is set to true, the metadata for field's metadata is updated on store.
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp]: What metadata and how and when they are updated?

File: include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp [hunk 30]
```diff
@@ -240,6 +328,10 @@ namespace SuperiorMySqlpp
 
 
         public:
+            /**
+             * Validates metadata for given ResultBindings compared to metadata recieved from the server.
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp:352]: Just mention please, when it is called.

**Review by majcl — CHANGES_REQUESTED** [review]
I like the documentation adding. I hope it will be really helpful in future development. But there is one thing i miss here: Somewhat thorough description of how ParamsBindings and ResultBindings works in PreparedStatements and DynamicPreparedsStatement. 
When the storage of provided arguments is used and when the provided argument's values are only copied.

File: include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp (hunk 27 above)
Comment (OpatrilPeter) [reply include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp:196]: This comment is deleted because this function was in fact not working at all - and is rewritten along with proper documentation in following pull request #63.

File: include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp (hunk 26 above)
Comment (OpatrilPeter) [reply include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp:219]: According to MySql documentation, yes. It calls `mysql_stmt_result_metadata` that fetches this information independently on `execute()`, see example at https://dev.mysql.com/doc/refman/5.7/en/mysql-stmt-fetch.html.

File: include/superior_mysqlpp/prepared_statement.hpp (hunk 8 above)
Comment (OpatrilPeter) [reply include/superior_mysqlpp/prepared_statement.hpp]: By empty I meant null -- I like to consider Nullable equivalent of Optionals. Do you have any better way to describe it in mind?

File: include/superior_mysqlpp/prepared_statement.hpp (hunk 10 above)
Comment (OpatrilPeter) [reply include/superior_mysqlpp/prepared_statement.hpp]: It is count indeed. IMO, because in C++ it is common that size() returns the amount of elements for std containers, I consider size and count to be equivalent terms. Are you considering ambiguity with `sizeof`-like semantics?

File: include/superior_mysqlpp/prepared_statement.hpp (hunk 11 above)
Comment (OpatrilPeter) [reply include/superior_mysqlpp/prepared_statement.hpp]: See preceding discussion.

File: include/superior_mysqlpp/prepared_statement.hpp (hunk 12 above)
Comment (OpatrilPeter) [reply include/superior_mysqlpp/prepared_statement.hpp]: Good catch! The actual order of operations here is following:
`mysql_stmt_execute()` <- this preforms the query, producing the result set (for SELECT-like statements)
*validation*
(optional step for `store()`) `mysql_stmt_store_result()` <- this, however, performs the "heavy lifting" of delivery

File: include/superior_mysqlpp/prepared_statements/binding_types.hpp (hunk 14 above)
Comment (OpatrilPeter) [reply include/superior_mysqlpp/prepared_statements/binding_types.hpp]: fixaed ;)

File: include/superior_mysqlpp/prepared_statements/get_binding_type.hpp (hunk 16 above)
Comment (OpatrilPeter) [reply include/superior_mysqlpp/prepared_statements/get_binding_type.hpp]: I haven't seen any specific support in C client for more exotic architectures, but I feel that being more explicit in data type choice (with no extra cost) is a defensive step in right direction nontheless -- I suggest leaving this one kept as it is - in following pull request, this code was rewritten, so we can debate over that specific change more concretely and productively.

File: include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp (hunk 23 above)
Comment (OpatrilPeter) [reply include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp]: Will clarify in more detailed description. FYI, the same method exist for subclass that has *use*() strategy and for subclass that does *store*() strategy - it does nothing for *use*, but calls mysql_stmt_store_result for *store*.

File: include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp (hunk 25 above)
Comment (OpatrilPeter) [reply include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp]: It is not guaranteed explicitly in documentation. Further research needs to be made into source code if such behaviour is explicitly mentioned at least there or if it is purely implemntation detail.

File: include/superior_mysqlpp/prepared_statement.hpp (hunk 8 above)
Comment (OpatrilPeter) [reply include/superior_mysqlpp/prepared_statement.hpp]: I clarified the meaning a little - mainly the fact that this flag is about storing results.

File: include/superior_mysqlpp/prepared_statements/default_initialize_result.hpp (hunk 15 above)
Comment (OpatrilPeter) [reply include/superior_mysqlpp/prepared_statements/default_initialize_result.hpp]: As far as I can see, this function itself is only called from function right below - most likely it should be inside detail:: namespace. That said, it actually calls specializable function InitializeResultItemImpl, which does nothing in general variant, but can be specialized for arbitrary type - this information is already provided in comment above.

File: include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp (hunk 28 above)
Comment (OpatrilPeter) [reply include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp:305]: After looking harder, this one seems problematic. According to https://dev.mysql.com/doc/refman/5.7/en/mysql-stmt-attr-set.html, it does following: Number of rows to fetch from server at a time _when using a cursor_.  MariaDB does not mention that at all. We don't even have required method (outside DBDriver) to create cursors.

For now, I'll only add commentary about this.

File: include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp (hunk 25 above)
Comment (OpatrilPeter) [reply include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp]: I did some testing, as basic tests did not cover this.
After calling this method in use() mode, it always returns 0 for me (with debian-jessie default mysql client). Was that ever tested at all?

I'll add suggestion for removal.

Comment (OpatrilPeter) [reply include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp]: Alternatively, correct implementation can be created by simple counter addition in that specialization.

File: include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp (hunk 24 above)
Comment (OpatrilPeter) [reply include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp]: I hope description of tparam for all related classes and in general usage description will be enough.

File: include/superior_mysqlpp/prepared_statements/validate_metadata_modes.hpp [hunk 31]
```diff
@@ -44,6 +50,12 @@ namespace SuperiorMySqlpp
         }
     }
 
+    /**
+     * Function returns for given field type "typeId" shared by same types
+     * @param type
+     * @return Arbitary integer that is however unique for each group of same types
+     * !!!!!
```
Comment (OpatrilPeter) [inline include/superior_mysqlpp/prepared_statements/validate_metadata_modes.hpp]: Request for feedback, this part is a guesswork.

File: include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp (hunk 21 above)
Comment (OpatrilPeter) [reply include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp:309]: I added this information to the general version.

File: include/superior_mysqlpp/prepared_statement.hpp [hunk 32]
```diff
@@ -56,19 +141,48 @@ namespace SuperiorMySqlpp
...
+     * ResultBindings specifies bindings that shall be used for query results.
+     * @tparam Types List of types matching query results.
+     */
     template<typename... Types>
     using ResultBindings = Bindings<false, Types...>;
 
-
+    /**
+     * @brief Static MySQL prepared statement.
+     * Static prepared statements needs to know the parameter bindings.
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statement.hpp]: Actually there is no need to have parameters, the query can be just one string as in normal query, but possible results must be bounded to some storage..

File: include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp (hunk 25 above)
Comment (majcl) [reply include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp]: Yes, this should probably be then counted manually in use spec.

**Review by majcl — APPROVED** [review]
